apps/designer/app/designer/features/style-panel/shared/use-style-data.ts

```diff
@@ -70,38 +80,59 @@ export const useStyleData = ({ selectedInstance, publish }: UseStyleData) => {
         return;
       }
 
-      store.createTransaction([stylesStore], (styles) => {
-        const instanceId = selectedInstance.id;
-        const breakpointId = selectedBreakpoint.id;
-
-        for (const update of updates) {
-          if (update.operation === "set") {
-            replaceByOrAppendMutable(
-              styles,
-              {
-                breakpointId,
-                instanceId,
-                property: update.property,
-                value: update.value,
-              },
-              (item) =>
-                item.instanceId === instanceId &&
-                item.breakpointId === breakpointId &&
-                item.property === update.property
-            );
+      store.createTransaction(
+        [styleSourceSelectionsStore, styleSourcesStore, stylesStore],
+        (styleSourceSelections, styleSources, styles) => {
+          const instanceId = selectedInstance.id;
+          const breakpointId = selectedBreakpoint.id;
+
+          // crate style source and its selection on currently selected instance
+          let matchedStyleSourceSelection = styleSourceSelections.find(
+            (styleSourceSelection) =>
+              styleSourceSelection.instanceId === instanceId
+          );
+          if (matchedStyleSourceSelection === undefined) {
+            const styleSourceId = ObjectID.toString();
```

Comment:
do we want to keep using ObjectID or switch to uuid?

Reply:
I think we should also switch uuid to https://twitter.com/_ericelliott/status/1611214704554442752?t=8DUheU9KGq7mtDENA-nlwA&s=19

Reply:
Looks like nanoid have most qualities
https://github.com/paralleldrive/cuid2#the-contenders